File: api/Models/VoucherType.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;

namespace api.Models
{
    [Index(nameof(Name), IsUnique = true)]
    public class VoucherType
    {
        public int Id { get; set; }
        [Required]
        [MaxLength(50)]
        public string Name { get; set; } = default!;
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public DateTime? DeletedAt { get; set; }
        public ICollection<VoucherSeries> VoucherSeries { get; set; } = new List<VoucherSeries>();
        public ICollection<Sale> Sales { get; set; } = new List<Sale>();
    }
}

File: api/app/models/voucher_type.ts
import { DateTime } from 'luxon';
import { BaseModel, column, hasMany } from '@adonisjs/lucid/orm';
import VoucherSerie from './voucher_serie.js';
import type { HasMany } from '@adonisjs/lucid/types/relations';
import Sale from './sale.js';

export default class VoucherType extends BaseModel {
  @column({ isPrimary: true })
  declare id: number;

  @column()
  declare name: string;

  @column.dateTime({ autoCreate: true })
  declare createdAt: DateTime;

  @column.dateTime({ autoCreate: true, autoUpdate: true })
  declare updatedAt: DateTime;

  @column.dateTime()
  declare deletedAt: DateTime | null;

  @hasMany(() => VoucherSerie)
  declare voucherSeries: HasMany<typeof VoucherSerie>;

  @hasMany(() => Sale)
  declare sales: HasMany<typeof Sale>;
}
